request_test.go

```diff
@@ -1237,6 +1238,76 @@ func TestRequest_BodyMultipart(t *testing.T) {
 		eof, _ := reader.NextPart()
 		assert.Nil(t, eof)
 	})
+
+	t.Run("multipart writer error", func(t *testing.T) {
+		cases := []struct {
+			name     string
+			reqFn    func() *Request
+			assertFn func(t *testing.T, req *Request)
+		}{
+			{
+				name: "with form",
+				reqFn: func() *Request {
+					req := NewRequestC(config, "POST", "url")
+					req.multipartFn = func(w io.Writer) *multipart.Writer {
+						return multipart.NewWriter(&failureWriter{})
+					}
+					req.WithMultipart()
+					req.WithForm(map[string]string{"foo": "bar"})
+					return req
+				},
+				assertFn: func(t *testing.T, req *Request) {
+					req.chain.assertFailed(t)
+				},
+			},
+			{
+				name: "with form field",
+				reqFn: func() *Request {
+					req := NewRequestC(config, "POST", "url")
+					req.multipartFn = func(w io.Writer) *multipart.Writer {
+						return multipart.NewWriter(&failureWriter{})
+					}
+					req.WithMultipart()
+					req.WithFormField("foo", "bar")
+					return req
+				},
+				assertFn: func(t *testing.T, req *Request) {
+					req.chain.assertFailed(t)
+				},
+			},
+			{
+				name: "with file",
+				reqFn: func() *Request {
+					req := NewRequestC(config, "POST", "url")
+					req.multipartFn = func(w io.Writer) *multipart.Writer {
+						return multipart.NewWriter(&failureWriter{})
+					}
+					req.WithMultipart()
+					req.WithFile("foo", "bar", strings.NewReader("baz"))
+					return req
+				},
+				assertFn: func(t *testing.T, req *Request) {
+					req.chain.assertFailed(t)
+				},
+			},
+		}
+
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				req := tc.reqFn()
+				tc.assertFn(t, req)
+			})
+		}
+	})
+}
+
+// failureWriter is a writer that returns error.
+type failureWriter struct {
+	io.Writer
+}
```

Comment:
Most our are in mocks_test.go; let's move this one there too?

We can rename it to "mockFailureWriter". Or we can make it more generic and name it "mockWriter", and add "err" field that is used like this:

```
func (fw *failureWriter) Write(p []byte) (n int, err error) {
    if fw.err != nil {
        return 0, fw.err
    }
    return fw.Writer.Write(p)
}
```

Reply:
ok will do